src/main/java/technology/semi/weaviate/client/base/http/impl/CommonsHttpClientImpl.java

```diff
@@ -39,6 +39,9 @@ public String getMethod() {
   @Override
   public HttpResponse sendGetRequest(String url) throws Exception {
     HttpGet httpGet = new HttpGet(url);
+    if (headers != null && headers.size() > 0) {
+      headers.forEach(httpGet::addHeader);
+    }
     httpGet.setHeader("Accept", "*/*");
     return sendRequest(httpGet);
```

Comment:
This code looks the same for each request. Does it make sense to factor it out into an extra method?